subsys/bluetooth/mesh/proxy.h

```diff
@@ -14,11 +14,47 @@
 int bt_mesh_pb_gatt_send(struct bt_conn *conn, struct net_buf_simple *buf,
 			 bt_gatt_complete_func_t end, void *user_data);
 
+#define PDU_TYPE(data)     (data[0] & BIT_MASK(6))
+#define CFG_FILTER_SET     0x00
+#define CFG_FILTER_ADD     0x01
+#define CFG_FILTER_REMOVE  0x02
+#define CFG_FILTER_STATUS  0x03
+
+#define PDU_HDR(sar, type) (sar << 6 | (type & BIT_MASK(6)))
+
+typedef int (*proxy_send_cb_t)(struct bt_conn *conn,
+			       const void *data, uint16_t len,
+			       bt_gatt_complete_func_t end, void *user_data);
+
+typedef void (*proxy_recv_cb_t)(struct bt_conn *conn,
+				struct bt_mesh_net_rx *rx,
+				struct net_buf_simple *buf);
+
+struct bt_mesh_proxy_role {
+	struct bt_conn *conn;
+	uint8_t msg_type;
+
+	struct {
+		proxy_send_cb_t send_cb;
+		proxy_recv_cb_t recv_cb;
+	} cb;
+
+	struct k_work_delayable sar_timer;
+	struct net_buf_simple buf;
+};
+
 #define BT_MESH_PROXY_NET_PDU   0x00
 #define BT_MESH_PROXY_BEACON    0x01
 #define BT_MESH_PROXY_CONFIG    0x02
 #define BT_MESH_PROXY_PROV      0x03
 
+ssize_t bt_mesh_proxy_msg_recv(struct bt_mesh_proxy_role *role,
+			       const void *buf, uint16_t len);
+int bt_mesh_proxy_msg_send(struct bt_mesh_proxy_role *role, uint8_t type,
+			   struct net_buf_simple *msg,
+			   bt_gatt_complete_func_t end, void *user_data);
+void bt_mesh_proxy_msg_init(struct bt_mesh_proxy_role *role);
+
 int bt_mesh_proxy_prov_enable(void);
 int bt_mesh_proxy_prov_disable(bool disconnect);
 
```

Comment:
This is a bit of a hassle, perhaps, but ideally, this header should be split up according to the new source file split, and the functions be named appropriately

Reply:
The file has been split, but it's better to change the name in a PR with the split services.